### PR TITLE
Feat: ArgoCD return values & pluginengine refactor

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

ArgoCD return values & pluginengine refactor

### CODE REVIEW

File: cmd/argocd/main.go [hunk 1]
```diff
@@ -12,23 +12,23 @@ const NAME = "argocd"
 type Plugin string
 
 // Install implements the installation of ArgoCD.
```
Comment (lfbdev) [inline cmd/argocd/main.go:14]: Install should be create

File: cmd/argocd/main.go [hunk 2]
```diff
@@ -12,23 +12,23 @@ const NAME = "argocd"
...
 }
 
 // Reinstall implements the reinstallation of ArgoCD.
-func (p Plugin) Reinstall(options *map[string]interface{}) (bool, error) {
-	return argocd.Reinstall(options)
+func (p Plugin) Update(options *map[string]interface{}) (map[string]interface{}, error) {
+	return argocd.Update(options)
 }
 
 // Uninstall implements the uninstallation of ArgoCD.
```
Comment (lfbdev) [inline cmd/argocd/main.go:24]: uninstall should be delete

File: cmd/devlake/main.go [hunk 3]
```diff
@@ -12,25 +12,25 @@ const NAME = "devlake"
...
+func (p Plugin) Update(options *map[string]interface{}) (bool, error) {
 	return devlake.Reinstall(options)
 }
 
-// Uninstall Uninstall the installation of DevLake.
-func (p Plugin) Uninstall(options *map[string]interface{}) (bool, error) {
-	return devlake.Uninstall(options)
-}
-
 // IsHealthy implements the healthy check of DevLake.
```
Comment (lfbdev) [inline cmd/devlake/main.go:24]: ishealthy shoud be read

**Comment by daniel-hutao** [comment]
close by #195

**Pull request closed without merging**